pennylane_lightning/core/utils/MPIManager.hpp

```diff
@@ -736,6 +739,30 @@ class MPIManager {
                                        recvtag, this->getComm(), &status));
     }
 
+    /**
+     * @brief MPI_GatherV wrapper.
+     *
+     * @tparam T C++ data type.
+     * @param recvBuf Receive buffer vector.
+     * @param sendBuf Send buffer vector.
+     * @param root Rank of destination.
+     * @param displacements Elements shifted from each rank for gather.
+     */
+    template <typename T>
+    void GatherV(std::vector<T> &sendBuf, std::vector<T> &recvBuf,
+                 std::size_t root, std::vector<int> &displacements) {
```

Comment:
```suggestion
                 std::size_t root, std::vector<std::size_t> &displacements) {
```
To keep consistency with the res of the code.

Reply:
the problem is that MPI requires an array of ints for displacements. For the other inputs e.g. root, it's just a single number that we can static_cast, but for this vector we will need to cast every element (e.g. with transform), which introduces unnecessary complexity, so I think it's ok to keep int here. This is the only MPI method used here that requires an array of integers